wyvern/components/models/base_ranking_model.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+import asyncio
+from functools import cached_property
+from typing import Dict, Generic, List, Optional, Set, Union
+
+from wyvern.components.models.model_component import (
+    GENERALIZED_WYVERN_ENTITY,
+    MODEL_OUTPUT_DATA_TYPE,
+    REQUEST_ENTITY,
+    ModelComponent,
+    ModelInput,
+    ModelOutput,
+    SingleEntityInput,
+)
+from wyvern.entities.identifier import Identifier
+
+
+class BaseRankingModel(
+    ModelComponent[
+        ModelInput[GENERALIZED_WYVERN_ENTITY, REQUEST_ENTITY],
+        ModelOutput[MODEL_OUTPUT_DATA_TYPE],
+    ],
+    Generic[GENERALIZED_WYVERN_ENTITY, REQUEST_ENTITY, MODEL_OUTPUT_DATA_TYPE],
+):
+    @cached_property
+    def manifest_feature_names(self) -> Set[str]:
+        return set()
+
+    async def single_inference(
+        self,
+        input: SingleEntityInput[GENERALIZED_WYVERN_ENTITY, REQUEST_ENTITY],
+    ) -> Union[float, str, List[float]]:
+        """
+        Single inference for ranking model
+        """
+        raise NotImplementedError
+
+    async def inference(
+        self,
+        input: ModelInput[GENERALIZED_WYVERN_ENTITY, REQUEST_ENTITY],
+        **kwargs,
+    ) -> ModelOutput[MODEL_OUTPUT_DATA_TYPE]:
+        """
+        Batch inference for ranking model
+        """
+        model_score_batches = await asyncio.gather(
+            *[
+                self.single_inference(
```

Comment:
I think this should be inverted haha. Single inference should call batch inference since presumably batch inference would be faster than calling one inference at a time

Reply:
I want the "async def inference" interface to be the only interface exposed by the base "ModelComponent".

I'll update. I think we don't need a "single_inference" interface for now

Reply:
I think it's worth having both, but ideally single would call batch by default